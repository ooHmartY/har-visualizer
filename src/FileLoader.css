.file-loader {
    color: white;
    height: 100%;
}

.file-loader__content {
    padding-top: 20%;
}

.file-loader__actions {
    margin-top: 15px;
}

.file-loader__upload-button {
    padding: 20px;
    border: 1px solid midnightblue;
    background-color: royalblue;
    border-radius: 5px;
    display: inline-block;
}

.file-loader__upload-input {
    display: none;
}
