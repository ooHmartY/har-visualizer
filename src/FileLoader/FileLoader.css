.file-loader {
    color: white;
    height: 100%;
}

.file-loader__content {
    padding-top: 20%;
    height: 100%;
    border: 5px solid transparent;
    border-radius: 15px;
    transition: transform 600ms ease-in-out;
}

.file-loader--over .file-loader__content {
    border-color: white;
    transform: scale(0.9);
    opacity: 0.8;
}

.file-loader__actions {
    margin-top: 15px;
}

.file-loader__upload-button {
    padding: 20px;
    border: 1px solid midnightblue;
    background-color: royalblue;
    border-radius: 5px;
    display: inline-block;
}

.file-loader__upload-input {
    display: none;
}
