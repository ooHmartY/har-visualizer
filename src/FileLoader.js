import React, { Component, PropTypes } from 'react';
import './FileLoader.css';
import fileReader from './utils/FileReader';

function first(list) {
    return (list && list.length && list[0]) || undefined;
}

export default class FileLoader extends Component {

    constructor(props) {
        super(props);
        this.onFileAdded.bind(this);
    }

    onFileAdded(e) {
        const { onFileAdded } = this.props;
        fileReader(first(e.target.files || e.dataTransfer.files))
            .then(data => {
                onFileAdded(data);
            });
    }

    render() {
        return (
            <div
                className="file-loader"
                onDrop={e => {
                    e.preventDefault();
                    this.onFileAdded.call(this, e);
                }}
                onDragOver={e => e.preventDefault()}
                onDragEnter={e => e.preventDefault()}
            >
                <div className="file-loader__content">
                    <div className="file-loader__copy">
                        Drag and Drop a File or Click Below to Upload
                    </div>
                    <div className="file-loader__actions">
                        <label className="file-loader__upload-button">
                        <span className="file-loader__upload-label">
                            Add File +
                        </span>
                            <input
                                type="file"
                                className="file-loader__upload-input"
                                onChange={this.onFileAdded.bind(this)} />
                        </label>
                    </div>
                </div>
            </div>
        );
    }
}

FileLoader.propTypes = {
    onFileAdded: PropTypes.func.isRequired
};

FileLoader.defaultProps = {
    onFileAdded: () => {}
};
