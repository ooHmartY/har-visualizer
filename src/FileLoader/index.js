import React, { Component } from 'react';
import PT from 'prop-types';
import autoBind from 'react-autobind';
import { head } from 'lodash';

import classNames from 'classnames';
import './FileLoader.css';
import fileReader from '../utils/FileReader';

export default class FileLoader extends Component {

    static propTypes = {
        onFileAdded: PT.func.isRequired,
    };

    constructor(props) {
        super(props);
        this.state = {
            isDragOver: false,
        };
        autoBind(this);
    }

    onFileAdded(e) {
        const { onFileAdded } = this.props;
        fileReader(head(e.target.files || e.dataTransfer.files))
            .then((data) => {
                onFileAdded(data);
            });
    }

    render() {
        const { isDragOver } = this.state;
        const className = classNames(
            'file-loader',
            {
                'file-loader--over': isDragOver,
            },
        );

        return (
          <div
            className={className}
            onDrop={(e) => {
                e.preventDefault();
                this.onFileAdded.call(this, e);
            }}
            onDragOver={e => e.preventDefault()}
            onDragEnter={(e) => {
                e.preventDefault();
                this.setState({
                    isDragOver: true,
                });
            }}
            onDragLeave={() => this.setState({
                isDragOver: false,
            })}
          >
            <div className="file-loader__content">
              <div className="file-loader__copy">
                  Drag and Drop a File or Click Below to Upload
               </div>
              <div className="file-loader__actions">
                <label
                  className="file-loader__upload-button"
                  htmlFor="fileInput"
                >
                  <span className="file-loader__upload-label">
                            Add File +
                        </span>
                  <input
                    type="file"
                    id="fileInput"
                    className="file-loader__upload-input"
                    onChange={this.onFileAdded}
                  />
                </label>
              </div>
            </div>
          </div>
        );
    }
}
