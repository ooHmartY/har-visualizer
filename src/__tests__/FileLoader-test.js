import React from 'react';
import FileLoader from '../FileLoader';
import renderer from 'react-test-renderer';

describe('FileLoader', () => {
    it('renders initial state correctly', () => {
        const component = renderer.create(<FileLoader/>);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
