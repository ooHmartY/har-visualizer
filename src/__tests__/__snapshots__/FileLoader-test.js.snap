exports[`FileLoader renders initial state correctly 1`] = `
<div
  className="file-loader"
  onDragEnter={[Function onDragEnter]}
  onDragOver={[Function onDragOver]}
  onDrop={[Function onDrop]}>
  <div
    className="file-loader__content">
    <div
      className="file-loader__copy">
      Drag and Drop a File or Click Below to Upload
    </div>
    <div
      className="file-loader__actions">
      <label
        className="file-loader__upload-button">
        <span
          className="file-loader__upload-label">
          Add File +
        </span>
        <input
          className="file-loader__upload-input"
          onChange={[Function bound onFileAdded]}
          type="file" />
      </label>
    </div>
  </div>
</div>
`;
